### DISCUSSION TITLE
Performance impact with multiple plugins ?

### DISCUSSION THREAD

**Discussion by flou** [apollographql/router#2291] [Q&A] [answered]
Hello all

I am wondering what the best way is to handle additional functionalities with the router.

Is it better to have:
- separate all into several small plugins
- one big plugin with all 

It's easier to organise code by splitting features into separate plugins but I wonder if there is a performance impact with having a lot of small plugins vs. a single plugin that does everything?

**Comment by bnjjj** [comment, marked as answer]
Having several plugins instead of just 1 won't really have a significant impact on performance. Just keep in mind to not have a plugin dependent of another one. Also there is no way to force ordering between plugins calls that's why it's better to not have dependencies between different plugins.
